micro-deps/micro-deps-spring-config/src/main/groovy/com/ofg/infrastructure/discovery/DependencyResolutionConfiguration.groovy

```diff
@@ -34,6 +35,9 @@ class DependencyResolutionConfiguration {
     @Autowired(required = false)
     private DependencyPresenceOnStartupVerifier dependencyPresenceOnStartupVerifier
 
+    @Autowired(required = false)
```

Comment:
Yes, this is how I'd like to give a user possibility to use custom implementation of the factory. There is no bean registered and auto wiring leaves the field empty, i.e. null, and the default implementation will be used. In case of registered bean this bean will be used.

Why not annotations? micro-deps has no dependency on Spring. I didn't want to introduce it because of this single annotation.